Sources/NIOPosix/System.swift

```diff
@@ -195,19 +220,32 @@ internal func syscall<T: FixedWidthInteger>(blocking: Bool,
     }
 }
 
-/* Sorry, we really try hard to not use underscored attributes. In this case however we seem to break the inlining threshold which makes a system call take twice the time, ie. we need this exception. */
+
+/*
+ * Sorry, we really try hard to not use underscored attributes. In this case
+ * however we seem to break the inlining threshold which makes a system call
+ * take twice the time, ie. we need this exception.
+ */
 @inline(__always)
-internal func wrapErrorIsNullReturnCall<T>(where function: String = #function, _ body: () throws -> T?) throws -> T {
+@discardableResult
+internal func sysReadCall<T: FixedWidthInteger>(where function: String = #function,
```

Comment:
Hmm, I don't love this: it's too specific to `read` (same with those above). Maybe `syscallForbiddingEINVAL`?